release_bot/releasebot.py

```diff
@@ -66,13 +66,44 @@ def load_release_conf(self):
         release_conf = self.conf.load_release_conf(conf)
         self.new_release.update(release_conf)
 
+    def process_version_from_title(self, title, latest_version):
+        """
+        checks for a valid version keyword, and if found
+        returns the requested release version
+        :param title: the pr/issue title to check keyword for
+        :param latest_version: the current latest version with type Version()
+        :return match: true if it is a valid version
+        :return version: version string for requested latest release   
```

Comment:
codacy complains about trailing whitespace here, please remove it

Reply:
okay! 
for the Test do I need to write a test that involves complete process from creating an a new issue named for eg: 'new major release' to checking whether the bot detects it does the release or I can just write a unit test for the specific funtion where I just manually send the current version and possible keywords and check for the results ? I think the former is better ?

Reply:
no need to write an end to end test. Just write a test where you pass several sets of inputs to the method process_version_from_title and check if you are getting the results you expect -- unit test is just fine